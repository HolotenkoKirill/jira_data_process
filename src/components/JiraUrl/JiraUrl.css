.jira-url-title {
	margin-top: 30px !important;
}

.jira-url-container {
	width: 400px;
	text-align: center;
	margin: 30px auto;
}
