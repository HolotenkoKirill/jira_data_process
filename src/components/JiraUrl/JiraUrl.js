import React, {Fragment} from 'react';
import TextField from '@material-ui/core/TextField';
import Typography from '@material-ui/core/Typography';

import './JiraUrl.css';

const JiraUrl = ({jiraUrl, isJiraUrlTouched, handleJiraUrlChange}) => {
    return (
        <Fragment>
            <Typography
                variant="h6"
                gutterBottom
                align="center"
                className="jira-url-title"
            >
                Please, enter JIRA URL below! (Example - jira.organization.com)
            </Typography>
            <form className="jira-url-container" noValidate autoComplete="off">
                <TextField
                    variant="outlined"
                    required
                    error={isJiraUrlTouched && !jiraUrl}
                    helperText="Jira URL field can't be empty."
                    label="Jira URL"
                    type="text"
                    defaultValue={jiraUrl}
                    onChange={handleJiraUrlChange}
                />
            </form>
        </Fragment>
    );
};

export default JiraUrl;
