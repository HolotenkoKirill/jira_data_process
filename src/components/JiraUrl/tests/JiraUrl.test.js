import React from 'react';
import {shallow} from 'enzyme';
import JiraUrl from '../JiraUrl';
import serializer from 'enzyme-to-json/serializer';

expect.addSnapshotSerializer(serializer);

describe('<JiraUrl />', () => {
    it('renders JiraUrl component', () => {
        const component = shallow(<JiraUrl jiraUrl={'jira.example.com'} />);

        expect(component).toMatchSnapshot();
    });

    it('should not show an error, when isJiraUrlTouched is true and jiraUrl prop is provided', () => {
        const component = shallow(
            <JiraUrl jiraUrl={'jira.example.com'} isJiraUrlTouched />
        );

        expect(
            component.find('WithStyles(ForwardRef(TextField))').prop('error')
        ).toEqual(false);
    });

    it('should show an error, when isJiraUrlTouched is true and jiraUrl prop is not provided', () => {
        const component = shallow(<JiraUrl isJiraUrlTouched />);

        expect(
            component.find('WithStyles(ForwardRef(TextField))').prop('error')
        ).toEqual(true);
    });
});
