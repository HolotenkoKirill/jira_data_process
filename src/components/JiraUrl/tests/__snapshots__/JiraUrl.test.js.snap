// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JiraUrl /> renders JiraUrl component 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Typography))
    align="center"
    className="jira-url-title"
    gutterBottom={true}
    variant="h6"
  >
    Please, enter JIRA URL below! (Example - jira.organization.com)
  </WithStyles(ForwardRef(Typography))>
  <form
    autoComplete="off"
    className="jira-url-container"
    noValidate={true}
  >
    <WithStyles(ForwardRef(TextField))
      defaultValue="jira.example.com"
      helperText="Jira URL field can't be empty."
      label="Jira URL"
      required={true}
      type="text"
      variant="outlined"
    />
  </form>
</Fragment>
`;
